torch/fx/experimental/proxy_tensor.py

```diff
@@ -303,6 +304,30 @@ def __torch_dispatch__(cls, func_overload, types, args=(), kwargs=None):
                     break
         assert proxy_mode is not None, "At least one argument must be a ProxyTensor"
 
+        if func_overload.__name__ == 'torch.cond':
+            # TODO: need to parse
+            print("In cond")
+            assert kwargs is None or not kwargs
+            pred, true_fn, false_fn, *operands = args
+            true_graph = get_isolated_graphmodule(true_fn, *operands, {})
+            false_graph = get_isolated_graphmodule(false_fn, *operands, {})
+            
+            # Note - all the fixed strings here will need random postfix/prefix to not collide, fine for now
+                
+            proxy_mode.tracer.root.register_module("true_graph", true_graph)
+            proxy_mode.tracer.root.register_module("false_graph", false_graph)
+
+            # For a smaller graph that esentially does the same thing, comment out these two lines and replace args with 
+            # true_graph / false_graph
+            # get_true_proxy = proxy_mode.tracer.create_proxy("get_attr", "true_graph", tuple(*operands), {})
+            # get_false_proxy = proxy_mode.tracer.create_proxy("get_attr", "false_graph", tuple(*operands), {})            
+            
+            args = (pred, true_graph, false_graph, *operands)
+            proxy_res = proxy_mode.tracer.create_proxy('call_function', func_overload, args, kwargs,
```

Comment:
If you extract the proxy attribute of ProxyTensor and pass that to `create_proxy` instead of args. Then you'll get the graph you want, without constant tensors and correctly propagated arguments.
```suggestion
            def unwrap_proxy(e):
                return e.proxy if isinstance(e, ProxyTensor) else e
            proxy_args = pytree.tree_map(unwrap_proxy, args)
            proxy_res = proxy_mode.tracer.create_proxy('call_function', func_overload, proxy_args, kwargs,
```
Then instead of returning just `proxy_res` we probably need to wrap it first back into ProxyTensor with `wrap_output`.

Reply:
Awesome, thank you.